src/app/pages/details/common/resource-box/resource-box-utils.tsx

```diff
@@ -32,7 +70,7 @@ function resourceBoxPermissions({resourceData, resourceStatus, loginUrl}) {
         locked: {
             iconType: 'lock',
             link: {
-                text: <FormattedMessage id='resources.loginToUnlock' />,
+                text: <FormattedMessage id="resources.loginToUnlock" /> as unknown as string,
```

Comment:
This feels a little dirty. I'd rather have a non-hook function to generate a string, but I don't find it in the package.

Reply:
Is this only used/displayed in `left-content.tsx`? What if you made the text value here the message id, then do something like `<FormattedMessage id={model.link.text} />` in that element, would that work?

Reply:
The `unlocked` entry also has `link.text` that doesn't use intl, so I'd need to make the recipient handle either `text` or `id` depending on what it's given.

Reply:
I made a LinkText component to handle it.

Reply:
Nice, I like that